### PR TITLE
Cannot trust or retire a retired node

### PR DESCRIPTION
**Pull request by jumaffre**
Follow up from https://github.com/microsoft/CCF/issues/778

It should not be possible to trust a node that is already retired. Also, the `ccf.nodes` tables should not be writable from Lua - we really want to limit what members can change in that table (i.e. only trust and retire nodes, via C++ functions).

### CODE REVIEW

**Comment by cimetrics** [comment]
cannot_trust_retire_node@[4558 aka 20200205.2](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=4558&view=results) vs master ewma over 30 builds from [4170](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=4170&view=results) to [4556](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=4556&view=results)
images

File: src/runtime_config/default_whitelists.h [hunk 1]
```diff
@@ -32,7 +32,6 @@ namespace ccf
       Tables::MEMBER_CERTS,
```
Comment (olgavrou) [inline src/runtime_config/default_whitelists.h]: Members should have access to the nodes table via lua, same as they do to the members table, etc. If they want to change the way that the nodes are handled they should be able to propose and vote for that.

Comment (achamayou) [reply src/runtime_config/default_whitelists.h]: The join protocol is implemented in C++ though, and we do want to enforce a particular set of conditions and transitions for nodes to preserve security guarantees. What's the use case for opening this up completely?

Comment (jumaffre) [reply src/runtime_config/default_whitelists.h]: Raised https://github.com/microsoft/CCF/issues/806

**Pull request merged**